### PR TITLE
VZ-10647 create boilerplate files for the Authproxy

### PR DESCRIPTION
**Pull request by davisgomes**
**Changes**
- Add new directory for the Authproxy
- Create boilerplate files for the initial setup of the Authproxy

### CODE REVIEW

File: authproxy/Makefile [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+GO_LDFLAGS ?= -extldflags -static -X main.buildVersion=${BUILDVERSION} -X main.buildDate=${BUILDDATE}
+
+.PHONY: build
+build: go-fmt go-vet
+	go build -o bin/manager main.go
+
+# Run against the configured Kubernetes cluster in ~/.kube/config
+.PHONY: run
+run:
+	# To enable webhooks: $(GO) run main.go --kubeconfig=${KUBECONFIG} --enable-webhooks=true --metrics-addr=localhost:0 --cert-dir=build/webhook-certs
```
Comment (fredtibbitts) [inline authproxy/Makefile:56]: Since Auth Proxy doesn't have webhooks this should be removed.

File: authproxy/Makefile [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	RELEASE_BRANCH=$(shell git rev-parse --abbrev-ref HEAD)
+endif
+
+DIST_DIR:=dist
+GO ?= CGO_ENABLED=0 GO111MODULE=on GOPRIVATE=github.com/verrazzano go
+GO_LDFLAGS ?= -extldflags -static -X main.buildVersion=${BUILDVERSION} -X main.buildDate=${BUILDDATE}
+
+.PHONY: build
+build: go-fmt go-vet
+	go build -o bin/manager main.go
```
Comment (fredtibbitts) [inline authproxy/Makefile:51]: I know this was just copied as all of the other Makefiles have this, but I wonder what this is even used for? There's a `go-build` target that seems to be more robust, I wonder if we use this anywhere?

**Review by fredtibbitts — APPROVED** [review]

File: authproxy/Makefile (hunk 2 above)
Comment (davisgomes) [reply authproxy/Makefile:51]: I'm not sure. this does the go fmt check and the go-vet (whatever that means). Not sure if it's totally necessary.

File: authproxy/Makefile (hunk 1 above)
Comment (davisgomes) [reply authproxy/Makefile:56]: Thanks. I missed this one.

**Pull request merged**